utils/funcexp/functor_str.h

```diff
@@ -104,6 +105,18 @@ class Func_Str : public Func
 				return fp->data()->getStrVal(row, isNull);
 			break;
 		}
+        // Strip off the '+' that %g puts before the exponent symbol
+        dest = src = buf;
+        while(*src != '\0')
+        {
```

Comment:
The problem is that the '+' is on the exponent and appears in the middle of the string.